### ISSUE TITLE
Web - The first letter of the chat name is cut

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22997]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any chat with other user
2. add "/settings" to the address in address bar (like "https://new.expensify.com/r/4148003058225040/settings")
3. Open Room Name
4. Notice that name is "hat Report", first letter is cut

## Expected Result:
he name should be the same as on previous page

## Actual Result:
The first letter of the chat name is cut

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.40-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/b2a8bda0-7413-4ece-a257-bf1329dffd8b

https://github.com/Expensify/App/assets/93399543/ad4d201a-d100-4a9e-8ca6-fb7b0ec49208

**Expensify/Expensify Issue URL:**
**Issue reported by:**  @alexxxwork
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689405309465269

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alexxxwork** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?
The root cause is that we strip first letter of chat report name for all chats (rooms and ordinary chats) here:
https://github.com/Expensify/App/blob/67bac9d50b9fbc1b8a1b1a14e8cc567ca3d0580b/src/libs/ReportUtils.js#L1207-L1210

### What changes do you think we should make in order to solve the problem?

We can add a condition to check whether the first letter is "#" or not

### What alternative solutions did you explore? (Optional)

**Comment by CortneyOfstad** [comment]
Was able to recreate so going to get eyes on this:

<img width="1345" alt="Screenshot 2023-07-17 at 10 31 45 AM" src="https://github.com/Expensify/App/assets/10926327/8bc79159-070b-4470-b4fb-4a79ede84ed0">

**Comment by CortneyOfstad** [comment]
@robertKozik We have a couple of proposals already shared above — feel free to share your thoughts/comments/etc when you can 👍

**Comment by makiour** [comment]
Contributor details
Your Expensify account email: m.akiour@aui.ma
Upwork Profile Link: https://www.upwork.com/freelancers/~0120be4cc910d2909e

**Comment by robertKozik** [comment]
@CortneyOfstad Sure thing, I'll try review existing proposals today

Hi @makiour It's good to have you here, this issue doesn't have any proposal selected yet,  so fell free to post your proposal as well.

**Comment by makiour** [comment]
Thank you @robertKozik, I will put my proposal in a bit! Thanks for your interaction.

**Comment by makiour** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When we try to check the settings page, and room name more specifically, we ran into cutting the first letter as you can see from the picture I provided:
![image](https://github.com/Expensify/App/assets/60530038/f0fd3dce-bd4a-4072-9aaa-7c0f10ef8b0f)

After testing it on my own account, it seems that it happens only at first time when we have a default value of Chat Report. 

### What is the root cause of that problem?

The root issue is coming from the first time we initiate creating a chat with a user. Mainly, if we check the network request with the onyxData, we can clearly see that once we change the room-name, the data is returned with a dash as you can see below:
![image](https://github.com/Expensify/App/assets/60530038/4da0bcd9-de4e-4ff3-8565-aadd6c6249f9)

The issue happens exactly when we just create an account (having that first chat with concierge) or when initiating a chat with any other user. As you can see the response is different and it is static to "Chat Report" without including a #dash.  
![image](https://github.com/Expensify/App/assets/60530038/72a4986f-899c-4c05-a69a-b3efcd82ce8c)

Thus, the issue doesn't have to go with changing the logic of deleting the first letter as we are guaranteed that the room name includes a dash once it is changed as the validation and custom creation forces including the dash 

### What changes do you think we should make in order to solve the problem?
The change I suggest:
Rather than changing the whole blocks where we eliminate the first letter, which will impact the different types of chats and already store report names, the simplest solution is to change the static name to include "#chat-report" since the beginning and causing everything to show as requested. 
When changing this line of code to "#chat-report", it will first comply to the validation (having no uppercase, numbers, etc.) ) 
https://github.com/Expensify/App/blob/9004c3968fcde7f46a473db871b4dd6ab2f3effc/src/CONST.js#L626C9-L626C28

Thus, we will be able to see the full name of the room when creating any chat with any user.

### What alternative solutions did you explore? (Optional)
The alternative solution is to change the logic of eliminating the first letter, which will need to check if the input includes a dash and differentiating between the two cases. Also, if this solution is the one followed, we need also to keep it in mind whenever a room name convention has changed or when dealing with dashes. Still, this will also fix the issue!

**Comment by robertKozik** [comment]
Thank you all for your proposals. I'll address each individually:

@ginsuma - I appreciate your [proposal](https://github.com/Expensify/App/issues/22997#issuecomment-1637922583) the most as it addresses the issue with accessing the settings page. However, it does prevent accessing the settings page for policy expense chats, even though it can be accessed in the "normal" way, to ensure consistency between accessing the settings page from the app and from the deep link. Please consider if wrapping them in FullPageNotFoundView is the best solution. If you can address the problem with policy expense chats, it would be a strong contender.

@alexxxwork - While your [proposal](https://github.com/Expensify/App/issues/22997#issuecomment-1637924100) should work fine, I believe it doesn't address the root cause of the problem. Trimming the first letter of the name is a workaround for accessing a screen that shouldn't be accessible for a particular chat.

@samh-nl - The second option in your [proposal](https://github.com/Expensify/App/issues/22997#issuecomment-1637970111) is moving in the right direction as the room name changing page shouldn't be accessible for every type of chat. However, I don't think changing the trimming itself is necessary. If you could expand the second option to the proposal, it'll be taken under consideration

@makiour - I don't think you correctly addressed the root of the problem since it doesn't only occur the first time the chat name value is default. The ability to change the room name should be limited for specific chats. Trimming works fine when it's applicable (when the screen can be accessed by clicking within the app).

To summarize, most of the proposals focus on patching the trimming mechanism rather than addressing the root cause of the problem. We should concentrate on prohibiting access to the screen via the deep link, ensuring consistency with what can be accessed from within the app. @ginsuma's proposal is a strong contender if it can also handle the issue with policy expense chats, which his proposal introduces.

**Comment by makiour** [comment]
Thank you for your comment @robertKozik 

Could you please mention any other cases when this happens as well? As the room name has a validation structure, it seems that we can never save without the dash whenever room it is changed. No matter if we limit changing room name or not, the trimming will always be fine. So the default value is the only one that serms to cause the issue. @alexxxwork, thank you for your like to the comment as well! Do you have anything to add here as I am unable to see another issue with room-name rather than that or the solution looks fine to you.

Thank you so much Robert for your time and comment once again. Happy to take this discussion further.

**Comment by robertKozik** [comment]
I understand your point, @makiour. Let me provide an example to illustrate the issue. In the case of a normal chat with one person, when you deep link to the settings page, it isn't aware of this type of chat and presents the user with the option to change the name, even though settings shouldn't be accessible. On the other hand, in a thread where the settings page is accessible by clicking within the app, deep linking won't cause this issue because the settings page handles the thread and doesn't present the option to change the name.

These examples demonstrate that the problem lies in the settings page not properly handling all types of chats, rather than the trimming itself. Changing the name works fine for group chats, and patching the name trimming would be over-engineering in this case, as it's not a supported option for threads or normal chats.

Considering these points, I would agree with the approach of fixing the trimming when the changing name field can be accessed through the normal flow of the app, but i think it's not.

**Comment by robertKozik** [comment]
Thanks @ginsuma for updating your proposal. 

Your proposal effectively addresses the root cause of the issue (also as first one states it correctly) by preventing deep linking to the settings page for chats where it shouldn't be accessible. I also like the idea of expand these checks to other settings pages too.

Due to these reasons, I'd go with @ginsuma [proposal](https://github.com/Expensify/App/issues/22997#issuecomment-1637922583)

🎀 👀 🎀 C+ reviewed

**Comment by arosiclair** [comment]
Excellent work managing these proposals @robertKozik 👍

**Comment by Pujan92** [comment]
Regression here https://github.com/Expensify/App/pull/23255#issuecomment-1645754623

**Comment by CortneyOfstad** [comment]
@robertKozik just bumping the check list above ^^ TIA!

**Comment by robertKozik** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [X]  [@robertKozik] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/23255
> * [X]  [@robertKozik] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:  https://github.com/Expensify/App/pull/23255#discussion_r1270818242
> * [X]  [@robertKozik] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
> * [X]  [@robertKozik] Determine if we should create a regression test for this bug. Don't think so, It was rather the fault of C+/C on the imprecise PR review process, as It was simple mistake.
> * [ ]  [@robertKozik] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
> * [ ]  [@CortneyOfstad] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by CortneyOfstad** [comment]
I'm heading OoO until 8/14 and since this requires payment tomorrow, so reassigning in the meantime. 

@alexxxwork — contract has been offered for reporting the issue
@ginsuma — has a contract already in place in Upwork; however, looking through the GH it shows that the assignment took place on July 18, PR was proposed on July 20, but there was a regression. Not sure if this qualifies for the bonus. Thoughts @arosiclair / @robertKozik?

**Comment by arosiclair** [comment]
> Not sure if this qualifies for the bonus. Thoughts @arosiclair / @robertKozik?

I guess the regression nullifies the bonus, but it might be worth asking in #bug-zero since I haven't seen this kind of thing happen before

**Comment by garrettmknight** [comment]
I'd agree that the regression nullifies the bonus. We want to reward quick, thorough work.

Paying out now!

**Comment by robertKozik** [comment]
Hey, want to just point out that I'm part of the external agency (Software Mansion) so I'm not eligible for payment

**Comment by garrettmknight** [comment]
Ah, nevermind then. Thanks for the heads up!
